src/mlpack/methods/reinforcement_learning/environment/continuous_double_pole_cart.hpp

```diff
@@ -0,0 +1,374 @@
+      l1(l1),
+      l2(l2),
+      gravity(gravity),
+      massCart(massCart),
+      forceMag(forceMag),
+      tau(tau),
+      thetaThresholdRadians(thetaThresholdRadians),
+      xThreshold(xThreshold),
+      doneReward(doneReward),
+      maxSteps(maxSteps),
+      stepsPerformed(0)
+  { /* Nothing to do here */ }
+
+  /**
+   * Dynamics of Continuous Double Pole Cart instance. Get reward and next
+   * state based on current state and current action.
+   *
+   * @param state The current state.
+   * @param action The current action.
+   * @param nextState The next state.
+   * @return reward, it's always 1.0.
+   */
+  double Sample(const State& state,
+                const Action& action,
+                State& nextState)
+  {
+    // Update the number of steps performed.
+    stepsPerformed++;
+
+    arma::vec dydx(6, arma::fill::zeros);
+    for (size_t i = 0; i < 2; i++)
+    {
+      dydx[0] = state.Velocity();
+      dydx[2] = state.AngularVelocity(1);
+      dydx[4] = state.AngularVelocity(2);
+    }
+    Dsdt(state, action, dydx);
+    RK4(state, action, dydx, nextState);
+
+
+    // Check if the episode has terminated.
+    bool done = IsTerminal(nextState);
+
+    // Do not reward agent if it failed.
+    if (done && maxSteps != 0 && stepsPerformed >= maxSteps)
+      return doneReward;
+    else if (done)
+      return 0;
+
+    /**
+     * When done is false, it means that the cartpole has fallen down.
+     * For this case the reward is 1.0.
+     */
+    return 1.0;
+  }
+
+  /**
+   * This is the ordinary differential equations required for estimation of
+   * nextState through RK4 method.
+   *
+   * @param state The current state.
+   * @param action The action taken.
+   * @param dydx The differential.
+   */
+  void Dsdt(const State& state,
+            const Action& action,
+            arma::vec& dydx)
+  {
+    double totalForce = action.action;
+    double totalMass = massCart;
+    double omega1 = state.AngularVelocity(1);
+    double omega2 = state.AngularVelocity(2);
+    double sinTheta1 = std::sin(state.Angle(1));
+    double sinTheta2 = std::sin(state.Angle(2));
+    double cosTheta1 = std::cos(state.Angle(1));
+    double cosTheta2 = std::cos(state.Angle(2));
+
+    // Calculate total effective force.
+    totalForce += m1 * l1 * omega1 * omega1 * sinTheta1 + 0.375 * m1 * gravity *
+        std::sin(2 * state.Angle(1));
+    totalForce += m2 * l2 * omega2 * omega2 * sinTheta1 + 0.375 * m2 * gravity *
+        std::sin(2 * state.Angle(2));
+
+    // Calculate total effective mass.
+    totalMass += m1 * (0.25 + 0.75 * sinTheta1 * sinTheta1);
+    totalMass += m2 * (0.25 + 0.75 * sinTheta2 * sinTheta2);
+
+    // Calculate acceleration.
+    double xAcc = totalForce / totalMass;
+
+    // Calculate angular acceleration.
+    double angAcc1 = -0.75 * (xAcc * cosTheta1 + gravity * sinTheta1) / l1;
+    double angAcc2 = -0.75 * (xAcc * cosTheta2 + gravity * sinTheta2) / l2;
+
+    dydx[1] = xAcc;
```

Comment:
We could also use ``dydx[1]``, ``dydx[3]`` and ``dydx[5]`` right away, to store the values.